editoast/src/views/infra/auto_fixes.rs

```diff
@@ -234,10 +266,18 @@ mod test {
             .to_request();
         assert_eq!(call_service(&app, req_del).await.status(), StatusCode::OK);
 
-        let req_fix = TestRequest::get()
-            .uri(format!("/infra/{small_infra_id}/auto_fixes").as_str())
-            .to_request();
-        let response = call_service(&app, req_fix).await;
+        let infra_errors_before_fix: PaginatedResponse<crate::views::infra::errors::InfraError> =
+            read_body_json(call_service(&app, errors_request(small_infra_id)).await).await;
+        // Check that some new issues appeared
+        assert!(infra_errors_before_fix.count > infra_errors_before_all.count);
+
+        // WHEN
+        let response = call_service(&app, auto_fixes_request(small_infra_id)).await;
+
+        // THEN
+        let infra_errors_after_fix: PaginatedResponse<crate::views::infra::errors::InfraError> =
+            read_body_json(call_service(&app, errors_request(small_infra_id)).await).await;
+        assert_eq!(infra_errors_after_fix, infra_errors_before_fix);
```

Comment:
Why are you checking specifically the error list before and after calling `auto_fixes`?
If we want to check that the infra didn't change calling the endpoint, we could check the railjson.

Reply:
Good idea, I didn't think of that.
Changed done : https://github.com/osrd-project/osrd/compare/4a2c2f22132823ff66b3b2fdc2a3cecee6a0ad5a..a84d1773a6e0c037dcb09809c70e810fcb8a41ce

Note that I used the `/railjson` endpoint.
And I kept the first check on /errors to check that small_infra is not (too) broken.

Reply:
After discussion, back to `/errors` check https://github.com/osrd-project/osrd/compare/a84d1773a6e0c037dcb09809c70e810fcb8a41ce..464d8123926af1aba98e3a2effd772b42947cd92.
Cause: PartialEq impl macro on Railjson is not accurate (sorting order matters for comparison, but is not meaningful).

Note that the sorting issue is more distant on errors check, but it may wrongly break PartialEq after error regeneration.